Include OpenLayers as node module instead of git submodule

adube opened the pull request:
In order to make the installation procedure much easier,
this patches makes openlayers no longer included as a git
submodule, but as a node module instead.

Also, the version of openlayers is upgraded to master
(2016-01-29), with the addition to one patch that prevents the
ol3-externs.js file to throw errors when used in the build.
That patch is currently under a PR:
https://github.com/openlayers/ol3/pull/4756

adube commented:
@alexbrault Would you please test this patch ?  You need:
- node v4.x +
- checkout the branch
- run `npm install`
- run `make dist`

I get an error the first time.  If you do too and you're able to understand why, a fix would be welcome.

You can use `nvm` to install multiple node versions.

adube commented:
https://github.com/openlayers/ol3/pull/4756 has been merged.  Now using the ol3 official `master` version. As soon as `3.14.0` is out, it will be important to use it.

adube commented:
@fipp Would you be interested in testing this PR ?  Just checkout the https://github.com/mapgears/ol3-google-maps/tree/upgrade-ol3 branch, then try:
- install node 4.+ (you can use nvm)
- run `npm install` at the root of the directory
- run `make dist`.  You may get an error the first time.  Run it again if it occurs

Let me know if you're willing to try this.

fipp commented:
Hi @adube. I'll test it right away and let you know.

fipp commented:
@adube: I succeeded, but I did indeed need to run `make dist` a couple of times. I used my existing ubuntu/trusty64 VM with java 1.8 and closure-util globally installed. I managed to build version `v0.3-21-g96f9dbd` of ol3-google-maps. Is there any easy way of finding out which version of openlayers i should use with this? Maybe this should be written in the head of the ol3gm.js and ol3gm-debug.js files?

**PS**: I also managed to build using my os x el capitan box. The same problem occured the first time I ran `make dist`:

```
ERR! compile ERROR - Cannot read: node_modules/openlayers/build/ol-externs.js
```

adube commented:
The actual version of ol3 to use with ol3-google-maps shouldn't matter much as ol3-google-maps doesn't contain a lot of code.  OL3 3.13 should work fine.

This is something we could add in the release notes, i.e. "tested/builded with ol3 v.3.x.x".

Thank you very much for your tests.  I'll merge this soon even with the error remaining.

adube committed:
Include OpenLayers as node module instead of git submodule

In order to make the installation procedure much easier,
this patches makes openlayers no longer included as a git
submodule, but as a node module instead.

Also, the version of openlayers has been upgraded to
3.13.1, with an additional patch:
https://github.com/adube/ol3/commit/0a51e4ac329fe936352001d3e64804522cb01fac

adube commented:
Here's a final report about this PR, now that it is officially ready to get merged.

OpenLayers 3 made some major changes to their library, which would break the current version of OLGM.  For that matter, the `master` branch can't be safely used as version, as I would have wanted.

Instead, I forked ol3 on Mapgears' account, created a `v3.13.1-olgm` branch there, which includes openlayers 3.13.1 and the patch that fixes the issue with the ol3-externs.js file: https://github.com/adube/ol3/commit/0a51e4ac329fe936352001d3e64804522cb01fac

Which means that this version of OLGM is compatible with v3.13.x, but not the soon coming v3.14.0 version of ol3.

Merging this, now.  Thank you very much @fipp for your time and patience.
